internal/lifecycle/readme_test.go

```diff
@@ -0,0 +1,438 @@
+package lifecycle
+
+import (
+	"bufio"
+	"errors"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func setEnv(e map[string]string) error {
+	for k, v := range e {
+		err := os.Setenv(k, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func unsetEnv(e map[string]string) error {
+	for k, _ := range e {
+		err := os.Unsetenv(k)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func equalError(a, b error) bool {
+	if a == nil {
+		return b == nil
+	}
+	if b == nil {
+		return a == nil
+	}
+	return errors.Is(a, b)
```

Comment:
hmm, I'm not sure what this is doing, but it seems iffy. What are you trying to do?

Reply:
Resolved, thank you.